File: Darts/src/burlton/dartzee/code/bean/PlayerTypeFilterPanel.java
package burlton.dartzee.code.bean;

import burlton.core.code.util.Debug;
import burlton.desktopcore.code.bean.RadioButtonPanel;

import javax.swing.*;

public class PlayerTypeFilterPanel extends RadioButtonPanel
{
	public PlayerTypeFilterPanel()
	{
		add(rdbtnAll);
		add(rdbtnHuman);
		add(rdbtnAi);
	}

	private final JRadioButton rdbtnAll = new JRadioButton("All");
	public final JRadioButton rdbtnHuman = new JRadioButton("Human");
	public final JRadioButton rdbtnAi = new JRadioButton("AI");
	
	public String getWhereSql()
	{
		if (rdbtnAll.isSelected())
		{
			return "";
		}
		else if (rdbtnHuman.isSelected())
		{
			return "Strategy = -1";
		}
		else if (rdbtnAi.isSelected())
		{
			return "Strategy > -1";
		}
		else
		{
			Debug.stackTrace("Unexpected setup - no radio buttons selected?");
			return "";
		}
	}
}

File: Darts/src/burlton/dartzee/code/bean/PlayerTypeFilterPanel.kt
package burlton.dartzee.code.bean

import burlton.desktopcore.code.bean.RadioButtonPanel
import javax.swing.JRadioButton

class PlayerTypeFilterPanel : RadioButtonPanel()
{
    val rdbtnAll = JRadioButton("All")
    val rdbtnHuman = JRadioButton("Human")
    val rdbtnAi = JRadioButton("AI")

    init
    {
        add(rdbtnAll)
        add(rdbtnHuman)
        add(rdbtnAi)
    }

    fun getWhereSql(): String
    {
        return when
        {
            rdbtnHuman.isSelected -> "Strategy = -1"
            rdbtnAi.isSelected -> "Strategy > -1"
            else -> ""
        }
    }
}
